src/main/java/io/toast/tk/maven/plugin/RunScriptsMojo.java

```diff
@@ -0,0 +1,167 @@
+        } catch (IOException e) {
+            getLog().error(e);
+        }
+    }
+
+    private void initPluginModules() {
+        PluginLoader loader = new PluginLoader(this.pluginDir);
+        this.pluginModules = pluginDir == null ? new Module[]{} : loader.collectGuiceModules(loader.loadPlugins(IAgentPlugin.class.getClassLoader()));
+    }
+
+    private void initReportOutputDir() throws IOException {
+        File file = new File(this.outputDirectory);
+        if(!file.exists()){
+            file.mkdir();
+        } else if(!file.isDirectory()){
+            throw new IOException(this.outputDirectory + "isn't a valid directory !");
+        }
+    }
+
+    private List<File> toFileList(FileSet scriptSet) throws IOException {
+        File directory = new File(scriptSet.getDirectory());
+        String includes = toString(scriptSet.getIncludes());
+        String excludes = toString(scriptSet.getExcludes());
+        return FileUtils.getFiles(directory, includes, excludes);
+    }
+
+    private static String toString(List<String> strings) {
+        StringBuilder sb = new StringBuilder();
+        for (String string : strings) {
+            if (sb.length() > 0)
+                sb.append(", ");
+            sb.append(string);
+        }
+        return sb.toString();
+    }
+
+    private List<ITestPage> getScripts(List<File> files) {
+        List<ITestPage> testScripts = new ArrayList<>();
+        TestParser parser = new TestParser();
+        files.forEach(p -> {
+            try{
+                List<String> scriptLines = FileHelper.getScript(new FileInputStream(p));
+                ITestPage testScript = parser.parse(scriptLines, p.getName().toString());
+                testScripts.add(testScript);
+            }catch(Exception e){
+                getLog().error("Unable to parse " + p.getName() + " !", e);
+            }
+        });
+        return testScripts;
+    }
+
+    private void executeScripts( List<ITestPage> testScripts) {
+        testScripts.forEach(script -> {
+            try {
+                run(script);
+            } catch (Exception e) {
+                getLog().error("Failed to execute " + script.getName() + " !", e);
+            }
+        });
+    }
+
+    protected ITestPage run(ITestPage testPage) throws IOException {
+        getLog().info("Agent plugin class loader: " + IAgentPlugin.class.getClassLoader());
+        this.testPageRunner = new TestPageRunner(this.outputDirectory, this.pluginModules);
+        return testPageRunner.runTestPage(testPage);
+    }
+
+    public static void addURL(URL u, ClassLoader classLoader) throws IOException {
+        URLClassLoader sysloader = (URLClassLoader) classLoader;
+        Class sysclass = URLClassLoader.class;
+        try {
+            Method method = sysclass.getDeclaredMethod("addURL", new Class[]{URL.class});
+            method.setAccessible(true);
+            method.invoke(sysloader, new Object[]{u});
+        } catch (Throwable t) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1181)